web-ng/src/app/components/map/map.component.ts

```diff
@@ -69,6 +76,7 @@ export class MapComponent implements AfterViewInit, OnDestroy {
   };
   private selectedMarker?: google.maps.Marker;
   private markers: google.maps.Marker[] = [];
+  private polygons: Polygon[] = [];
```

Comment:
If you make this `Map<string, Polygon>` you won't need to iterate to find a polygon by id.

Reply:
Polygon changed to google.maps.Polygon

Reply:
> If you make this Map<string, Polygon> you won't need to iterate to find a polygon by id.

👆 

Could you also do this for `markers`?

Reply:
updated